packages/auth/graphql/resolvers/_mutations/addUserToRole.js

```diff
@@ -4,20 +4,28 @@ const transformUser = require('../../../lib/transformUser')
 const {
   ensureUserHasRole,
   userHasRole,
-  addUserToRole
+  addUserToRole,
+  isRoleClaimableByMe
 } = require('../../../lib/Roles')
-const logger = console
 
-module.exports = async (_, args, { pgdb, req, signInHooks }) => {
-  ensureUserHasRole(req.user, 'admin')
-  const { userId, role } = args
-  const user = await pgdb.public.users.findOne({id: userId})
+module.exports = async (_, args, { pgdb, signInHooks, user: me }) => {
+  const {
+    userId = me && me.id,
+    role
+  } = args
 
+  if (!(me && me.id === userId && isRoleClaimableByMe(role, me))) {
+    ensureUserHasRole(me, 'admin')
+  }
+
+  const user = await pgdb.public.users.findOne({ id: userId })
```

Comment:
Only fetch user from database if `me` is not user we ought to update. (And at this point we're pretty sure `me` is a user object already.)

```suggestion
  const user = me.id !== userId
    ? await pgdb.public.users.findOne({ id: userId })
    : me
```